tree/30_generic_methods/service_check_running_src.cf

```diff
@@ -0,0 +1,56 @@
+#####################################################################################
+# Copyright 2015 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name Service check running with SRC
+# @description Check if a service is running using SRC (AIX System Resource Controller)
+#
+# @parameter service_name Subsystem name in SRC
+#
+# @class_prefix service_check_running
+# @class_parameter service_name
+# This bundle will define a class service_check_running_${service_name}_{kept,not_ok}
+
+bundle agent service_check_running_src(service_name)
```

Comment:
Well, it could, but my vision of ncf is that we should be working the opposite way. One of our core principles is KISS - keep it simple. Merging this code into the upper bundle means that bundle will quickly become unwieldy and complex. What usually happens first in this kind of "mega-bundle" is that class conditions become very complex, and then we start making mistakes.

For an example, take a look at the ```service_action``` method, where we have merged all possible implementations into one method, in particular this class condition:

```
systemctl_utility_present|service_utility_present|init_d_directory_present|startsrc_utility_present|(windows.is_valid_action)::
```

Or this one in ```service_check_started_at_boot```:

```
pass2.(!(systemctl_utility_present|chkconfig_utility_present|lsitab_utility_present)|broken_systemctl)::
```

I want to avoid this as far as possible.

I see a much simple approach where we have a generic method for each type of implementation (src, systemd, upstart, init.d, ...), and ```service_check_running ``` is simply a "router" to choose which implementation to use according to the OS we're on and the utilities available.